parser/parse_doc.m

```diff
@@ -0,0 +1,88 @@
+function docStruct = parse_doc(functionPath)
+%PARSE_DOC Generates a structure from the input function's documentation.
+%
+% SYNTAX
+%
+%   docStruct = parse_doc(functionPath)
+%
+% DESCRIPTION
+%
+%   Parses the function corresponding to the input path and fetches the
+%   information corresponding to the function's documentation before
+%   reorganising it as a structure
+%
+% INPUTS
+%
+%   functionPath
+%     Character array corresponding to the path of the function.
+%
+% OUTPUTS
+%
+%   docStruct
+%     Structure containing the different parts of the function's
+%     documentation as its fields (summary, description, inputs, outputs,
+%     and notes).
+%
+% NOTES
+%
+% It requires a total respect of the template (e.g no "forgotten" spaces).
+% All the fields (SYNTAX, DESCRIPTION, etc...) must be provided in the
+% parsed function.
+
+%% Read the function and keep only the description section
+functionTxt = fopen(functionPath); % Open the function file
+
+fgetl(functionTxt); % Skip the first line (function...)
+% Initialize a cell that will receive the lines in the description
+functionDoc = [];
+
+textLine = fgetl(functionTxt);
+nLine = 1;
+headersPos = []; % Store the positions of the headers
+while length(textLine) >= 1 % Parsing will stop at first empty line
+    if erase(textLine, '%') ~= 0 % Delete the empty commented lines
+        
+        functionDoc = [functionDoc string(erase(textLine, '%'))];
+        
+        if sum(isstrprop(strip(erase(textLine, '%')), 'upper')) ==...
+                length(strip(erase(textLine, '%'))) % Check uppercase
+            
+            headersPos = [headersPos nLine]; % Store the header line
+            
+        end
+        nLine = nLine+1;
+    end
+    textLine = fgetl(functionTxt); % Reads the next line
+end
+
+fclose(functionTxt); % Close the function file
+headersPos = [headersPos length(functionDoc)+1]; % Position of the last line
+%% Fetch the function's summary
+docStruct.summary = functionDoc(1); % Store the summary of the function
+
+%% Fetch the other sections
+
+for nSection = 1:length(headersPos)-1 % Number of sections
+    header = strip(functionDoc(headersPos(nSection))); % Section name
+    sectionStart = headersPos(nSection)+1; % First line after header
+    sectionEnd = headersPos(nSection+1)-1; % Last line before next header
+    
+    switch header
+        case 'SYNTAX'
+            docStruct.syntax = erase(functionDoc(sectionStart:sectionEnd),'   ');
```

Comment:
The code is really whitespace dependent and the problem is that there is a lot of characters that are refered as whitespaces (tab, space, null, ...). When someone will use a different number of spaces, everything will break. A good way of reducing this problem and making the documentation more flexible is adding a symbol in front of specific information ('->' in front of inputs, '<- in front of outputs', '@' in front of notes). This way you know that everything that is after a certain symbol is a certian type until you reach the next symbol.

Reply:
Yes you're right but is it really easier to make sure everyone adds the right symbol in the right place or that they just respect the number of white spaces ? That's something I've been asking myself when I was coding but I don't have a clear preference for any of these options so I'm open to discussion.
Also, the way the code is implemented right now, there's no need for a symbol in `description` and `notes` because the code will just take everything that's in these sections and store it as a paragraph. I also don't see a genuine utility in having different symbols for the inputs and outputs as they are in different sections and thus can't be mixed up. But it might however be more visual to have these different arrows.

Reply:
The problem with the whitespaces is actually the implementation (tabs vs spaces vs null characters, etc.). There's a lot of ways to add space in a text editor. However with a symbol it's easy to know which one. 
The question of someone putting the right character is not relevant since it is the same for every way we find (number of spaces, etc.).
you are right for the description and notes, but I'm talking more about the input/outputs.